gfapi: Implement glfs_renameat()

vatsa287 opened the pull request:
Updates: #2717
Sponsored-By: iXsystems, Inc https://www.ixsystems.com/
Depends-On: #3541
Signed-off-by: Shree Vatsa N vatsa@kadalu.tech

aravindavk committed:
gfapi: Implement glfs_openat()

Example:

```c
const char *buff = "Hello World!";

fd1 = glfs_open(fs, "/", O_PATH);
fd2 = glfs_openat(fd1, filename, O_RDWR, 0);
glfs_write(fd2, buff, strlen(buff), flags);
```

Updates: #2717
Sponsored-By: iXsystems, Inc <https://www.ixsystems.com>
Change-Id: I4f8ad20616587f2d6e5feba4869abec96706a514
Signed-off-by: Aravinda Vishwanathapura <aravinda@kadalu.tech>

gluster-ant commented:
Can one of the admins verify this patch?

gluster-ant commented:
Can one of the admins verify this patch?

gluster-ant commented:
Can one of the admins verify this patch?

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/api/src/glfs-fops.c b/api/src/glfs-fops.c
index d9ba1aa51..ea19b95b7 100644
--- a/api/src/glfs-fops.c
+++ b/api/src/glfs-fops.c
@@ -3306,7 +3306,8 @@ invalid_fs:
 
 GFAPI_SYMVER_PUBLIC_DEFAULT(glfs_renameat, 11.0)
 int
-pub_glfs_renameat(struct glfs_fd *pglfd, const char *oldpath, const char *newpath, int flags)
+pub_glfs_renameat(struct glfs_fd *pglfd, const char *oldpath,
+                  const char *newpath, int flags)
 {
     int ret = -1;
     xlator_t *subvol = NULL;
diff --git a/api/src/glfs.h b/api/src/glfs.h
index beb8ed1b6..d17eae37e 100644
--- a/api/src/glfs.h
+++ b/api/src/glfs.h
@@ -864,8 +864,8 @@ glfs_rename(glfs_t *fs, const char *oldpath, const char *newpath) __THROW
     GFAPI_PUBLIC(glfs_rename, 3.4.0);
 
 int
-glfs_renameat(struct glfs_fd *glfd, const char *oldpath, const char *newpath, int flags) __THROW
-    GFAPI_PUBLIC(glfs_renameat, 11.0);
+glfs_renameat(struct glfs_fd *glfd, const char *oldpath, const char *newpath,
+              int flags) __THROW GFAPI_PUBLIC(glfs_renameat, 11.0);
 
 int
 glfs_link(glfs_t *fs, const char *oldpath, const char *newpath) __THROW
```

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/api/src/glfs.h b/api/src/glfs.h
index 39256a292..b17fe5c8b 100644
--- a/api/src/glfs.h
+++ b/api/src/glfs.h
@@ -864,8 +864,8 @@ glfs_rename(glfs_t *fs, const char *oldpath, const char *newpath) __THROW
     GFAPI_PUBLIC(glfs_rename, 3.4.0);
 
 int
-glfs_renameat(struct glfs_fd *glfd, const char *oldpath, const char *newpath)
-              __THROW GFAPI_PUBLIC(glfs_renameat, 11.0);
+glfs_renameat(struct glfs_fd *glfd, const char *oldpath,
+              const char *newpath) __THROW GFAPI_PUBLIC(glfs_renameat, 11.0);
 
 int
 glfs_link(glfs_t *fs, const char *oldpath, const char *newpath) __THROW
diff --git a/tests/basic/gfapi/gfapi-o-path.c b/tests/basic/gfapi/gfapi-o-path.c
index f05cf97d7..5bd8779e6 100644
--- a/tests/basic/gfapi/gfapi-o-path.c
+++ b/tests/basic/gfapi/gfapi-o-path.c
@@ -95,9 +95,11 @@ main(int argc, char *argv[])
     ret = glfs_renameat(fd1, filename, filename_renameat);
     VALIDATE_AND_GOTO_LABEL_ON_ERROR("glfs_renameat", ret, out);
 
-    if (glfs_access(fs, filepath, F_OK) == 0 || glfs_access(fs, filepath_renameat, F_OK) == -1){
+    if (glfs_access(fs, filepath, F_OK) == 0 ||
+        glfs_access(fs, filepath_renameat, F_OK) == -1) {
         ret = -1;
-        VALIDATE_AND_GOTO_LABEL_ON_ERROR("glfs_renameat operation has failed", ret, out);
+        VALIDATE_AND_GOTO_LABEL_ON_ERROR("glfs_renameat operation has failed",
+                                         ret, out);
     }
 
     ret = 0;
```

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/tests/basic/gfapi/gfapi-o-path.c b/tests/basic/gfapi/gfapi-o-path.c
index 744eb4be3..5bd8779e6 100644
--- a/tests/basic/gfapi/gfapi-o-path.c
+++ b/tests/basic/gfapi/gfapi-o-path.c
@@ -96,7 +96,7 @@ main(int argc, char *argv[])
     VALIDATE_AND_GOTO_LABEL_ON_ERROR("glfs_renameat", ret, out);
 
     if (glfs_access(fs, filepath, F_OK) == 0 ||
-        glfs_access(fs, filepath_renameat, F_OK) == -1){
+        glfs_access(fs, filepath_renameat, F_OK) == -1) {
         ret = -1;
         VALIDATE_AND_GOTO_LABEL_ON_ERROR("glfs_renameat operation has failed",
                                          ret, out);
```

vatsa287 committed:
gfapi: Implement glfs_renameat

Updates: gluster#2717
Sponsored-By: iXsystems, Inc <https://www.ixsystems.com>
Depends-On: #3541
Signed-off-by: Shree Vatsa N <vatsa@kadalu.tech>
